src/System.Private.Uri/src/Resources/Strings.resx

```diff
@@ -258,4 +258,13 @@
   <data name="IO_PathTooLong" xml:space="preserve">
     <value>The specified file name or path is too long, or a component of the specified path is too long.</value>
   </data>
+  <data name="Argument_ExtraNotValid" xml:space="preserve">
+    <value>Extra portion of URI not valid</value>
+  </data>
+  <data name="Argument_InvalidUriPart" xml:space="preserve">
+    <value>Uri part is not is not valid.</value>
```

Comment:
"Part" is not a word or concept that is exposed in the public API.  So, the error message needs to be something like this:

"The subcomponent, {0}, of this uri is not valid.".  

Note the use of {0} which means you should pass in the invalid string as a parameter so that the complete error message has meaningful data.